backend/service/fcc/fcc.go

```diff
@@ -0,0 +1,158 @@
+package fcc
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	v1 "github.com/ekspand/trusty/api/v1"
+	"github.com/ekspand/trusty/pkg/gserver"
+	"github.com/go-phorce/dolly/rest"
+	"github.com/go-phorce/dolly/xhttp/httperror"
+	"github.com/go-phorce/dolly/xhttp/marshal"
+	"github.com/go-phorce/dolly/xhttp/retriable"
+	"github.com/go-phorce/dolly/xlog"
+)
+
+// ServiceName provides the Service Name for this package
+const ServiceName = "fcc"
+
+var logger = xlog.NewPackageLogger("github.com/ekspand/trusty/backend/service", "fcc")
+
+const (
+	fccDefaultBaseURL = "https://apps.fcc.gov"
+)
+
+// Service defines the Status service
+type Service struct {
+	server *gserver.Server
+
+	FccBaseURL string
+}
+
+// Factory returns a factory of the service
+func Factory(server *gserver.Server) interface{} {
+	if server == nil {
+		logger.Panic("status.Factory: invalid parameter")
+	}
+
+	return func() {
+		svc := &Service{
+			server: server,
+		}
+
+		svc.FccBaseURL = fccDefaultBaseURL
+
+		server.AddService(svc)
+	}
+}
+
+// Name returns the service name
+func (s *Service) Name() string {
+	return ServiceName
+}
+
+// IsReady indicates that the service is ready to serve its end-points
+func (s *Service) IsReady() bool {
+	return true
+}
+
+// Close the subservices and it's resources
+func (s *Service) Close() {
+	logger.KV(xlog.INFO, "closed", ServiceName)
+}
+
+// RegisterRoute adds the Status API endpoints to the overall URL router
+func (s *Service) RegisterRoute(r rest.Router) {
+	r.GET(v1.PathForFccFrn, s.FccFrnHandler())
+	r.GET(v1.PathForFccSearchDetail, s.FccSearchDetailHandler())
+}
+
+type fccResponseWriter struct {
+	data []byte
+}
+
+func (w *fccResponseWriter) Write(data []byte) (int, error) {
+	w.data = append(w.data, data...)
+	return len(data), nil
+}
+
+// FccFrnHandler handles v1.PathForFccFrn
+func (s *Service) FccFrnHandler() rest.Handle {
+	return func(w http.ResponseWriter, r *http.Request, _ rest.Params) {
+		filerID, ok := r.URL.Query()["filer_id"]
+		if !ok || len(filerID) != 1 || filerID[0] == "" {
+			marshal.WriteJSON(w, r, httperror.WithInvalidRequest("missing filer_id parameter"))
+			return
+		}
+
+		httpClient := retriable.New()
```

Comment:
let's move this to a helper function in a pkg maybe.
so could be unittestable without Rest service.

we also should intro an interface, so we can mock and test this service